src/graphql-server/index.js

```diff
@@ -0,0 +1,67 @@
+
+const { ApolloServer } = require('apollo-server-express')
+const { ApolloServerPluginLandingPageGraphQLPlayground } = require('apollo-server-core')
+
+const http = require('http')
+const express = require('express')
+const cors = require('cors')
+const { promisify } = require('util')
+
+const Schema = require('../graphql-schema')
+
+module.exports = function GraphqlServer (ssb) {
+  return async function graphqlServer (opts, cb) {
+    if (cb === undefined) return promisify(graphqlServer)(opts)
+
+    const { port } = opts
+    const { typeDefs, resolvers } = Schema(ssb)
+
+    const app = express()
+
+    const graphiqlOrigin = `http://localhost:${port}`
+    const devServerOrigin = `http://localhost:${process.env.DEV_SERVER_PORT || 3000}`
+
+    app.use(cors({
+      origin (origin, cb) {
+        switch (process.env.NODE_ENV) {
+          case undefined: // if NODE_ENV not set, apply production standards
+          case 'production':
+            if (origin === undefined) return cb(null, true)
+            if (origin === graphiqlOrigin) return cb(null, true)
+            return cb(new Error('CORS issue')) // give no hints
+
+          case 'development':
+          case 'test':
+            if (origin === undefined) return cb(null, true)
+            if (origin === graphiqlOrigin) return cb(null, true)
+            if (origin === devServerOrigin) return cb(null, true)
+            return cb(new Error(`CORS issue - unexpected header 'Origin': '${origin}'`))
+
+          default:
+            cb(new Error(`invalid NODE_ENV - ${process.env.NODE_ENV}`))
+        }
+      }
+    }))
```

Comment:
I would disable all this CORS for the moment - it's not necessarily needed, and also that config is tuned for the Electron context (maybe)